Doc/library/ast.rst

```diff
@@ -146,6 +146,98 @@ Node classes
     Snakes <https://greentreesnakes.readthedocs.io/en/latest/>`__ project and
     all its contributors.
 
+
+.. _ast-root-nodes:
+
+Root nodes
+^^^^^^^^^^
+
+.. class:: Module(body, type_ignores)
+
+   A Python module.
+
+   *body* is a :class:`list` of the module's :ref:`ast-statements`.
+
+   *type_ignores* is a :class:`list` of the module's type ignore comments;
+   see :func:`ast.parse` for more details.
+
+   .. doctest::
+
+        >>> print(ast.dump(ast.parse('x = 1'), indent=4))
+        Module(
+            body=[
+                Assign(
+                    targets=[
+                        Name(id='x', ctx=Store())],
+                    value=Constant(value=1))],
+            type_ignores=[])
+
+
+.. class:: Expression(body)
+
+   A single Python expression.
```

Comment:
Can we also mention that this node type will be generated when the mode `eval` is passed to `ast.parse` for helping people figure out what is the difference among the root nodes?

Reply:
FWIW, the example does show this, though no harm mentioning it explicitly, I suppose.

Reply:
If we're going to do this here, might also make sense to be consistent and do it for the rest, no? Or is there something special about `Expression`/`eval`? Personally, I found `Interactive`/`single` to be the most non-obvious of the bunch, since there's no obvious connection between those two words.

In fact, I somehow completely missed seeing it myself in the example when I was testing something in connection with your comment below, thus being a counterpoint to my own here :joy: (though that also had to do with the valid values of the *mode* argument to `ast.parse` not being documented).

Reply:
Yep, I think we should mention it both places. The general mode is well known but function type / eval / single is not really known.

Reply:
```suggestion
   A single Python :ref:`expression input <expression-input>`.
   Node type generated by :func:`ast.parse` when *mode* is ``"eval"``.
```

As suggested, added here and elsewhere.

Also, clarify and cross-reference terminology here, as was done previously for interactive input.